server-spi/src/main/java/org/keycloak/organization/OrganizationProvider.java

```diff
@@ -139,8 +142,32 @@ default Stream<OrganizationModel> getAllStream() {
      *
      * @param organization the organization
      * @return Stream of the members. Never returns {@code null}.
+     * @deprecated Use {@link #getMembersStream(OrganizationModel, Map, Boolean, Integer, Integer)} instead.
      */
-    Stream<UserModel> getMembersStream(OrganizationModel organization, String search, Boolean exact, Integer first, Integer max);
+    @Deprecated(forRemoval = true, since = "26")
+    default Stream<UserModel> getMembersStream(OrganizationModel organization, String search, Boolean exact, Integer first, Integer max) {
+        Map<String, String> filters = Optional.ofNullable(search)
+                .map(value -> Map.of(UserModel.SEARCH, value))
+                .orElse(Map.of());
+        return getMembersStream(organization, filters, exact, first, max);
+    }
+
+    /**
+     * Returns the members of a given {@link OrganizationModel} filtered according to the specified {@code filters}.
+     *
+     * @param organization the organization
+     * @return Stream of the members. Never returns {@code null}.
+     */
+    default Stream<UserModel> getMembersStream(OrganizationModel organization, Map<String, String> filters, Boolean exact, Integer first, Integer max) {
+        var result = getMembersStream(organization, Optional.ofNullable(filters).orElse(Map.of()).get(UserModel.SEARCH), exact, first, max);
+        var membershipType = Optional.ofNullable(filters.get(MembershipType.NAME)).map(MembershipType::valueOf).orElse(null);
+
+        if (membershipType != null) {
+            return result.filter(userModel -> MembershipType.MANAGED.equals(membershipType) ? isManagedMember(organization, userModel) : !isManagedMember(organization, userModel));
+        }
+
+        return result;
+    }
```

Comment:
Not sure I like these defaults - even though they are overridden, the default impls are calling each other leading possibly to a stack overflow?

Reply:
True, done.